helix-lsp/src/client.rs

```diff
@@ -645,7 +655,14 @@ impl Client {
         text_document: lsp::TextDocumentIdentifier,
         position: lsp::Position,
         work_done_token: Option<lsp::ProgressToken>,
-    ) -> impl Future<Output = Result<Value>> {
+    ) -> Option<impl Future<Output = Result<Value>>> {
+        let capabilities = self.capabilities.get().unwrap();
+
+        #[allow(clippy::question_mark)]
+        if capabilities.signature_help_provider.is_none() {
+            return None;
+        }
```

Comment:
use the question mark and add a comment right before? "return early if signature help isn't supported"